server/application-server/src/main/java/de/tum/in/www1/hephaestus/syncing/NatsConsumerService.java

```diff
@@ -175,7 +178,7 @@ private String[] getSubjects() {
                 .map(String::toLowerCase)
                 .toArray(String[]::new);
 
-        return Arrays.stream(repositoriesToMonitor)
+        return adminService.getAdminConfig().getRepositoriesToMonitor().stream()
```

Comment:
The changes in `NatsConsumerService` are not sufficient to for NATS to send messages for dynamically changed repos during runtime. It runs once on startup and any further changes adding or removing a repo to monitor will not work as expected. 

I think that we have to use `consumerContext.getConsumerInfo().getConsumerConfiguration()` and update the `filterSubjects` once the monitored repos change with `streamContext.createOrUpdateConsumer`.

Alternatively, we can also move this dynamic configuration part of repos to monitor to a follow-up PR and remove it from this PR.

Reply:
Good point! Implemented this in d8cd507.

I added the `AdminConfigChangedEvent`, which gets emitted when the repositories are updated. The `NatsConsumerService` now listens for this event and updates the consumer accordingly. Unfortunately, I am getting NATS timeouts when updating the consumer - not sure if it's a Problem with my code or the server?

Reply:
I can try to take over this part 👍